### PR TITLE
Refactor/fix pre commit

### PR DESCRIPTION
**Pull request by grothe6**
New PR for issue https://github.com/qutech/QuMADA/issues/7: Minimium required refactoring for getting pre-commit working in the whole code base.

As discussed in https://github.com/qutech/QuMADA/pull/11, we start over with a fresh branch.

### CODE REVIEW

**Comment by grothe6** [comment]
@newton-per-sqm Refactoring for the updated code base was pretty straight forward. Pre-commit completes without errors now.

**Comment by newton-per-sqm** [comment]
> @newton-per-sqm Refactoring for the updated code base was pretty straight forward. Pre-commit completes without errors now.

I was expecting this, still haven't had time for the new PR yet. Thank you 😅!

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -29,4 +29,4 @@ repos:
     hooks:
     -   id: black
         args: ["--line-length", "120"]
-        language_version: python3.9
+        language_version: python3.11
```
[the remark this answers is not in the document]
Comment (grothe6) [reply .pre-commit-config.yaml:32]: I don't really know, what difference it makes, but in https://github.com/psf/black/blob/main/docs/integrations/source_version_control.md the documentation suggests using the latest supported python version.

[the remark this answers is not in the document]
Comment (grothe6) [reply .pre-commit-config.yaml:32]: Apparently, it doesn't really matter, see https://stackoverflow.com/a/69465958.

**Review by newton-per-sqm — APPROVED** [review]
Ok 👍

**Pull request merged**